packages/smoldot-provider/src/Database.ts

```diff
@@ -1,4 +1,53 @@
-export default interface Database {
+import * as pkg from '../package.json';
+
+// REM: Don't know how to make typescript happy. This doesn't work:
+// let create: (name: string) => Database | undefined = undefined;
+let create: any = undefined;
+
+if (typeof window === 'object') {
```

Comment:
Shouldn't this be wrapped in an async function? I'm surprised await is working without async here